rts/System/float3.cpp

```diff
@@ -11,6 +14,19 @@ CR_REG_METADATA(float3, (CR_MEMBER(x), CR_MEMBER(y), CR_MEMBER(z)))
 float float3::maxxpos = -1.0f;
 float float3::maxzpos = -1.0f;
 
+float3 float3::PickNonParallel() const
+{
+	// https://math.stackexchange.com/questions/3122010/how-to-deterministically-pick-a-vector-that-is-guaranteed-to-be-non-parallel-to
+	auto mi = std::min_element(std::begin(xyz), std::end(xyz), [](const auto& a, const auto& b) { return math::fabs(a) < math::fabs(b); });
+	auto Mi = std::max_element(std::begin(xyz), std::end(xyz), [](const auto& a, const auto& b) { return math::fabs(a) > math::fabs(b); });
```

Comment:
Why the opposite comparator? Won't it this result in `mi` and `Mi` being the same?

Reply:
Oh lol, you're right, nice catch!